platforms/documentation/docs/src/docs/userguide/best-practices/bp_tasks.adoc

```diff
@@ -123,4 +123,75 @@ include::sample[dir="snippets/bestPractices/useCacheabilityAnnotations/groovy",f
 - <<build_cache_concepts.adoc#non_cacheable_tasks,Non-cacheable Tasks>>
 
 === Tags
-`<<tags_reference.adoc#tag:tasks,#tasks>>`, `<<tags_reference.adoc#tag:caching,#caching>>`
+`#tasks`, `#caching`
+
+== Don't resolve Configurations before Task Execution
+
+Resolving configuration before the task execution phase can lead to incorrect results and slower builds.
+
+=== Explanation
+ 
+Calling manual resolution via link:{javadocPath}/org/gradle/api/artifacts/Configuration.html#resolve()[resolve] method returns a set of Files that doesn't preserve references to the tasks that produced these files.
+This will break automatic task dependencies wiring through task input properties.
+
+Resolving configuration outside task execution phase can slow down configuration phase even for tasks (e.g. `help`) that don't need results of the resolved configuration.
+
+=== Example
+
+==== Don't Do This
+
+====
+include::sample[dir="snippets/bestPractices/avoidResolvingConfigurationsManually/kotlin",files="library/build.gradle.kts;build.gradle.kts[tags=bad-classpath-printer]"]
+include::sample[dir="snippets/bestPractices/avoidResolvingConfigurationsManually/groovy",files="library/build.gradle;build.gradle[tags=bad-classpath-printer]"]
+====
+
+<1> *Add project dependency*: It implies that `:library:jar` task should be executed in order to resolve runtime classpath.
+<2> *Declare input files property as Set of files*: It doesn't track task dependencies.
+<3> *Pass resolved configuration to input files property*: Dependency on `:library:jar` task will be lost here.
+<4> *Consume actual jar file or classes directory to calculate digest*: Due to broken task dependencies, library.jar file will not exist at this point.
+
+==== Do This Instead
+
+Use lazy link:{javadocPath}/org/gradle/api/file/FileCollection.html[FileCollection] API instead.
+
+====
+include::sample[dir="snippets/bestPractices/avoidResolvingConfigurationsManually/kotlin",files="library/build.gradle.kts;build.gradle.kts[tags=good-classpath-printer]"]
+include::sample[dir="snippets/bestPractices/avoidResolvingConfigurationsManually/groovy",files="library/build.gradle;build.gradle[tags=good-classpath-printer]"]
+====
+
+<1> *Add project dependency*: It implies that jar task should be executed in order to resolve runtime classpath.
+<2> *Declare input files property as ConfigurableFileCollection*: It will track task dependencies.
+<3> *Pass configuration as-is to input files property*: It will be resolved when necessary and task dependencies will be preserved.
+<4> *Consume actual jar file or classes directory to calculate digest*: Here all files should exist at this point.
+
+=== Tags
+`#tasks`, `#inputs-and-outputs`, `#configurations`
```

Comment:
Do you mind updating this to linkable versions: `<<tags_reference.adoc#tag:tasks,#tasks>>`?

Reply:
Sure, I'll update it accordingly